j5/backends/hardware/sr/v4/power_board.py

```diff
@@ -143,7 +130,8 @@ def __init__(self, usb_device: usb1.USBDevice):
 
     @handle_usb_error
     def _read(self, command: ReadCommand) -> bytes:
-        return self._usb_handle.controlRead(0x80, 64, 0, command.code, command.data_len)
+        request_type = (0x80 & ~0x80) | 0x80
```

Comment:
From https://github.com/vpelletier/python-libusb1/blob/a90474b3388c3e9c6fb8788f2fdcb56980862e6b/usb1/__init__.py#L1497

Reply:
Happy to swap these out. I actually have little idea what it does, but it's some kind of bitmasking.

Maybe we should name the magic numbers?

Reply:
`request_type` is the `bmRequestType` of a setup packet (part of a control transfer), details of which can be found [here](https://www.beyondlogic.org/usbnutshell/usb6.shtml).

The bit-manipulation ops in the code you linked look like they're there to ensure the direction bit (bit 7) is set appropriately when the direction of the control transfer is known (i.e. when the user has chosen to call `controlWrite` or `controlRead`). As a result, `controlRead(0x00, ...)` doesn't accidentally initiate a write transfer, because the request type `0x00` is corrected to `0x80`.

I don't think this needs to be explicit here. I can see some benefit in naming the constant instead, e.g.

```python
CONTROL_REQUEST_TYPE_WRITE = 0x00
CONTROL_REQUEST_TYPE_READ = 0x80
```

but given that these names don't really add any extra information about what the value of the constant means, and that making this change would isolate the value of the constant from the place where its meaning is relevant (making it a bit harder to figure out what the code is doing), I don't think naming the constants would be a good idea.

Reply:
To clarify - I think passing `0x00` and `0x80` directly to `ctrl_transfer` is the least confusing option here (other options increase cognitive noise without increasing clarity).

Reply:
Will do.